app/js/lib/foxbox/foxbox.js

```diff
@@ -64,28 +64,27 @@ export default class Foxbox extends Service {
   init() {
     window.foxbox = this;
 
+    // No need to block the UI on the discovery process.
+    this._initDiscovery().then(() => {
+      return this[p.net].init();
+    })
+    .then(() => {
+      // Start polling.
+      this[p.settings].on('pollingEnabled', () => {
+        this.togglePolling(this[p.settings].pollingEnabled);
+      });
+      this.togglePolling(this[p.settings].pollingEnabled);
+
+      this.recipes = new Recipes({
+        settings: this[p.settings],
+        net: this[p.net]
+      });
+    });
+
     return this._initUserSession()
       .then(() => {
```

Comment:
Can you give more details about why moving the init user session there?
This methods contains the code that redirects the UI and needs to happen earlier, before setting polling and recipes at least.